README.md

```diff
@@ -1,7 +1,77 @@
 # Grateful Together API
 
+## Abstract: 
+This is the Backend API for the [Grateful Together App](https://www.FIX-THIS.com). Grateful Together is an application that allows groups of people to keep a central, easy to use, daily gratitude journal.
 
-### JSON Contract
+### Preview of App:
+Live Deployment link
+[Frontend Repo](https://github.com/GratefulTogether/gratefultogether-ui) 
+
+Add GIF here
+
+### Context:
+Collaborative capstone fullstack project for Mod 4 of Turing School of Software and Design.
+
+### Contributors
+<table>
+  <tr>
+    <td> Kailey Kaes: <a href="https://github.com/kaileykaes">Github</td>
+    <td> Jesse Sorman: <a href="https://github.com/jesse193">Github</td>
+    <td> K.D. Hubbard: <a href="https://github.com/kdhubb">Github</td>
+    <td> Steven Stackhouse: <a href="https://github.com/stackmm">Github</td>
+  </tr>
+</table>
+
+### Learning Goals:
+
+Develop a fullstack application in collaboration with a frontend team. Specifically, the backend team was tasked with creating a RESTful API service for the Grateful Together application, implementing at least one new technology. Our team also learned and executed an agile process from planning to development, through production, as well as a CI/CD process using CircleCI. 
+
+New Technologies implemented:
+- ActionCable
+- WebSockets
+- Action-Cable-Testing
+- CI/CD with CircleCI
+
+### Tech Stack
+- Ruby, Rails, WebSockets, ActiveRecord, PostgreSQL, Action-Cable-Testing, SimpleCov, RSpec, CI/CD with CircleCI
+
+## Usage
+
+#### Installation Instructions
+In order to run this application on your local machine, you'll need to have Ruby version 3.2.2, Rails 7.0.2.2 and PostgreSQL 14. 
+
+1. Clone this repo by executing: 
+```
+git clone git@github.com:GratefulTogether/gratefultogether-api.git 
+```
+    
+2. Install Gems:
+``` 
+bundle install 
+```
+
+3. Create and seed the development database:
+``` 
+rails db:{create,migrate,seed} 
+```
+
+#### Endpoints
+Returns all gratitude entries for today: 
+```
+GET /api/v1/wins
+```
+
+Returns all gratitude entries for a specific date (add the date in query parameters):
+```
+GET /api/v1/wins?date=2023-08-31
+```
+
+Post a new gratitude entry: 
+```
+POST /api/v1/wins
+```
+
+#### JSON Contract
```

Comment:
I would also be nice to see some documentation for what happens when things go wrong (errors). That will be useful for the FE, or anyone who consumes your API, to know what to expect when things error so they can handle accordingly.

Reply:
Just to verify understanding, for each one of our endpoints, should we describe all of the error handling we've written?

Reply:
We added an error response to the Readme for an invalid date for a POST request. We discussed about an error response for a get request, but are not sure about what error we would respond with. This is due to our get request are gotten by selecting a date from a built in calendar which automatically sends a valid date from the FE.

Reply:
I might to a 404 or perhaps a 400 (bad request) if for some reason a bad date gets through (ex: April 31 or some other string that isn't a date). Technically you KNOW you're front end app is going to send, presumably, a good date through. But what if someone else clones this BE app to make their own front end and they code it differently? I hear you that it's probably not a super high concern or even a must have but it is the kind of thing that makes documentation more complete. Because IF a bad date gets through in the request, what happens?